core/src/test/java/org/apache/iceberg/TestRewriteFiles.java

```diff
@@ -86,7 +86,7 @@ public void testEmptyTable() {
                     .rewriteFiles(
                         ImmutableSet.of(),
                         ImmutableSet.of(FILE_A_DELETES),
-                        ImmutableSet.of(FILE_A),
```

Comment:
Was it ever a valid use case?

Reply:
This would be removing a delete file and adding a data file and a new delete file? Seems impossible in rewrite

Reply:
Exactly, this is testing an error message so it was, probably, overlooked.